example/app.go

```diff
@@ -66,43 +62,59 @@ func Run() error {
 	evmListener := listener.NewEVMListener(ethClient, eventHandler, common.HexToAddress(ethCfg.SharedEVMConfig.Bridge))
 	mh := voter.NewEVMMessageHandler(ethClient, common.HexToAddress(ethCfg.SharedEVMConfig.Bridge))
 	mh.RegisterMessageHandler(common.HexToAddress(ethCfg.SharedEVMConfig.Erc20Handler), voter.ERC20MessageHandler)
-	evmVoter := voter.NewVoter(mh, ethClient)
-	evmChain := evm.NewEVMChain(evmListener, evmVoter, db, *ethCfg.SharedEVMConfig.GeneralChainConfig.Id, &ethCfg.SharedEVMConfig)
-
-	subC := subModule.NewSubstrateClient(stopChn)
-	err = subC.Configurate(viper.GetString(config.ConfigFlagName), "subConfig")
-	if err != nil {
-		panic(err)
-	}
-	subCfg := subC.GetConfig()
 
-	subL := subListener.NewSubstrateListener(subC)
-	subW := subWriter.NewSubstrateWriter(1, subC)
+	txFabric := evmtransaction.NewTransaction
 
-	// TODO: really not need this dynamic handler assignment
-	subL.RegisterSubscription(relayer.FungibleTransfer, subListener.FungibleTransferHandler)
-	subL.RegisterSubscription(relayer.GenericTransfer, subListener.GenericTransferHandler)
-	subL.RegisterSubscription(relayer.NonFungibleTransfer, subListener.NonFungibleTransferHandler)
-
-	subW.RegisterHandler(relayer.FungibleTransfer, subWriter.CreateFungibleProposal)
-	subChain := substrate.NewSubstrateChain(subL, subW, db, *subCfg.SharedSubstrateConfig.GeneralChainConfig.Id, &subCfg.SharedSubstrateConfig)
-
-	// Celo setup
-	ethClientCelo := evmclient.NewEVMClient()
+	evmVoter := voter.NewVoter(mh, ethClient, txFabric)
+	evmChain := evm.NewEVMChain(evmListener, evmVoter, db, *ethCfg.SharedEVMConfig.GeneralChainConfig.Id, &ethCfg.SharedEVMConfig)
+	// subC := subModule.NewSubstrateClient(stopChn)
```

Comment:
Aside from the comments, we're probably not going to merge these `app.go` changes into `main` as they are pretty specific to kaleido development. However, the `test_cli.sh` will be necessary merge in. Perhaps we could separate the PRs. It will probably be good to have a purely evm <-> evm, evm <-> substrate, etc. branch stashed away for development as well.

Also, the comm

Reply:
Yeah, I pushed a bunch of Kaleido-related code not relevant for this branch while testing. Going to make a new branch specific to kaleido (besu)